transit/script/bart.py

```
@@ -0,0 +1,396 @@
+
+    # Add columns and rearrange columns
+    BART["Key"] = BART["Station"] + BART["TOD"]
+    BART.columns = [
+        "Node",
+        "TOD",
+        "Boardings",
+        "Node Name",
+        "Station",
+        "Alightings",
+        "Key",
+    ]
+    BART = BART[
+        ["Node", "Node Name", "Station", "TOD", "Key", "Boardings", "Alightings"]
+    ]
+
+    # Sort and reset index
+    BART = BART[["Station", "TOD", "Key", "Boardings", "Alightings"]]
+    BART = BART.sort_values(by="Key").reset_index(drop=True)
+    BART.to_csv(output_dir / model_BART, index=False)
+    return BART
+
+
+# Function to map 'Station' to 'County'
+def map_station_to_county(station):
+    counties = {
+        "San Francisco": [
+            "EMBR",
+            "CIVC",
+            "24TH",
+            "MONT",
+            "POWL",
+            "GLEN",
+            "16TH",
+            "BALB",
+        ],
+        "San Mateo": ["DALY", "COLM", "SSAN", "SBRN", "SFIA", "MLBR"],
+        "Contra Costa": [
+            "RICH",
+            "ORIN",
+            "LAFY",
+            "WCRK",
+            "CONC",
+            "NCON",
+            "PITT",
+            "ANTC",
+            "DELN",
+            "PHIL",
+            "PCTR",
+            "PLZA",
+        ],
+        "Alameda": [
+            "WOAK",
+            "12TH",
+            "19TH",
+            "MCAR",
+            "ASHB",
+            "DUBL",
+            "WDUB",
+            "CAST",
+            "WARM",
+            "UCTY",
+            "SHAY",
+            "HAYW",
+            "BAYF",
+            "SANL",
+            "OAKL",
+            "COLS",
+            "FTVL",
+            "LAKE",
+            "ROCK",
+            "DBRK",
+            "NBRK",
+            "FRMT",
+        ],
+        "Santa Clara": [],  # Add stations for Santa Clara if available
+    }
+    for county, stations in counties.items():
+        if station in stations:
+            return county
+    return None
```

if we're not making the above changes to invert the dict first (which would then raise a KeyError if it's not found), then should this "return None" raise a Warning or an Exception instead of just returning None?